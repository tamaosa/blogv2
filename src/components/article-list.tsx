import React from "react"

import { ArticleLink } from "./article-link"
import { Mdx } from "../types/mdx"

type Props = {
  posts: Array<Mdx<"title" | "published" | "updated" | "tags">>
}

export const ArticleList: React.FC<Props> = ({ posts }) => {
  return (
    <ol style={{ listStyle: `none` }}>
      {posts.map(post => {
        return (
          <li key={post.fields.slug}>
            <ArticleLink {...post} />
          </li>
        )
      })}
    </ol>
  )
}
