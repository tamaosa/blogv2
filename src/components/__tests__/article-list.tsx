import React from "react"
import { render, screen } from "@testing-library/react"
import { ArticleList } from "../article-list"

describe("ArticleList", () => {
  const post = (slug: string) => {
    return {
      id: "id",
      excerpt: "excerpt",
      body: "body",
      frontmatter: {
        title: "title",
        published: "2022-01-01",
        updated: "2022-12-31",
        tags: ["tag1", "tag2", "tag3", "tag4", "tag5"],
      },
      fields: {
        slug: slug,
      },
    }
  }

  it("snapshot test1", () => {
    const { asFragment } = render(<ArticleList posts={[]} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("snapshot test2", () => {
    const { asFragment } = render(
      <ArticleList
        posts={[post("/slug1/"), post("/slug2/"), post("/slug3/")]}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("testing no elements", () => {
    render(<ArticleList posts={[]} />)
    const posts = screen.queryAllByTestId("article-link")
    expect(posts.length).toEqual(0)
  })

  it("testing number of elements", () => {
    render(
      <ArticleList
        posts={[post("/slug1/"), post("/slug2/"), post("/slug3/")]}
      />
    )
    const posts = screen.queryAllByTestId("article-link")
    expect(posts.length).toEqual(3)
  })
})
