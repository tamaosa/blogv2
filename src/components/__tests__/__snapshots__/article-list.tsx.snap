// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList snapshot test1 1`] = `
<DocumentFragment>
  <ol
    style="list-style: none;"
  />
</DocumentFragment>
`;

exports[`ArticleList snapshot test2 1`] = `
<DocumentFragment>
  <ol
    style="list-style: none;"
  >
    <li
      data-testid="article-link"
    >
      <article>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <a
            aria-label="記事へのリンク"
            href="/slug1/"
          >
            title
          </a>
        </div>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <div>
            <span
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              <time>
                2022年1月1日
              </time>
               公開
            </span>
            <span
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
               
              <time>
                2022年12月31日
              </time>
               更新
            </span>
          </div>
          <div>
            <div>
              <ul
                css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
              >
                <li>
                  <a
                    href="/tags/tag1/"
                  >
                    #tag1
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag2/"
                  >
                    #tag2
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag3/"
                  >
                    #tag3
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag4/"
                  >
                    #tag4
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag5/"
                  >
                    #tag5
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
    </li>
    <li
      data-testid="article-link"
    >
      <article>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <a
            aria-label="記事へのリンク"
            href="/slug2/"
          >
            title
          </a>
        </div>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <div>
            <span
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              <time>
                2022年1月1日
              </time>
               公開
            </span>
            <span
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
               
              <time>
                2022年12月31日
              </time>
               更新
            </span>
          </div>
          <div>
            <div>
              <ul
                css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
              >
                <li>
                  <a
                    href="/tags/tag1/"
                  >
                    #tag1
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag2/"
                  >
                    #tag2
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag3/"
                  >
                    #tag3
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag4/"
                  >
                    #tag4
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag5/"
                  >
                    #tag5
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
    </li>
    <li
      data-testid="article-link"
    >
      <article>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <a
            aria-label="記事へのリンク"
            href="/slug3/"
          >
            title
          </a>
        </div>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <div>
            <span
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              <time>
                2022年1月1日
              </time>
               公開
            </span>
            <span
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
               
              <time>
                2022年12月31日
              </time>
               更新
            </span>
          </div>
          <div>
            <div>
              <ul
                css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
              >
                <li>
                  <a
                    href="/tags/tag1/"
                  >
                    #tag1
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag2/"
                  >
                    #tag2
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag3/"
                  >
                    #tag3
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag4/"
                  >
                    #tag4
                  </a>
                </li>
                <li>
                  <a
                    href="/tags/tag5/"
                  >
                    #tag5
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
    </li>
  </ol>
</DocumentFragment>
`;
